src/tool/GenerateAction.cpp

```diff
@@ -20,11 +21,43 @@
 #include <mrdocs/Support/Path.hpp>
 #include <clang/Tooling/JSONCompilationDatabase.h>
 
+
 #include <cstdlib>
 
 namespace clang {
 namespace mrdocs {
 
+Expected<std::string>
+generateCompilationDatabaseIfNeeded(llvm::StringRef projectPath)
+{
+    namespace fs = llvm::sys::fs;
+    namespace path = llvm::sys::path;
+
+    fs::file_status fileStatus;
+    if (auto ec = fs::status(projectPath, fileStatus))
+    {
+        return Unexpected(Error(ec));
+    }
+
+    if (fs::is_directory(fileStatus))
+    {
+        return executeCmakeExportCompileCommands(projectPath);
+    }
+    else if (fs::is_regular_file(fileStatus))
```

Comment:
Else not needed because first conditionals return.

Not sure if `fs::is_regular_file` is needed at this point. We already checked if it's a directory. We don't need to check if it's a symlink.